src/main/java/com/mongodb/kafka/connect/source/schema/BsonValueToSchemaAndValue.java

```diff
@@ -173,20 +202,31 @@ private SchemaAndValue stringToSchemaAndValue(final Schema schema, final BsonVal
   }
 
   private SchemaAndValue bytesToSchemaAndValue(final Schema schema, final BsonValue bsonValue) {
-    byte[] value;
+    Object value = null;
     switch (bsonValue.getBsonType()) {
       case STRING:
         value = bsonValue.asString().getValue().getBytes(StandardCharsets.UTF_8);
         break;
       case BINARY:
         value = bsonValue.asBinary().getData();
         break;
+      case DECIMAL128:
```

Comment:
Why does this metho handle decimal128  but not other numeric types?

Reply:
This is due to how kafka has support for big decimals, via its logical types and for big decimals they are stored as a bytes.

Reply:
Ack